### PR TITLE
plugins.tvp: sport.tvp.pl handling

### PR DESCRIPTION
**Pull request by czarek7711**
Hello,

I've added handling of sport.tvp.pl live videos to the tvp plugin.

Admittedly I have little Python experience, but I tested the changes editing \pkgs\streamlink\plugins\_plugins.json, \pkgs\streamlink\plugins\tvp.py while removing the cached compiled file.

### CODE REVIEW

File: src/streamlink/plugins/tvp.py [hunk 1]
```diff
@@ -34,6 +35,9 @@
 @pluginmatcher(name="tvp_info", pattern=re.compile(
     r"https?://(?:www\.)?tvp\.info/",
 ))
+@pluginmatcher(name="tvp_sport", pattern=re.compile(
+    r"https?://sport\.tvp\.pl/(?P<stream_id>\d+)/.+$",
```
Comment (bastimeyer) [inline src/streamlink/plugins/tvp.py:39]:
```suggestion
    r"https?://sport\.tvp\.pl/(?P<stream_id>\d+)/.+",
```

**Review by bastimeyer — CHANGES_REQUESTED** [review]
Thanks for the PR.

Please post debug logs of a few streams. This stite is geo-blocked and inaccessible from the VPNs I have access to.

> editing \pkgs\streamlink\plugins_plugins.json

This is now how this works. You are not supposed to edit any Python distributions ("packages") directly. This will break things.

You need to install Streamlink in a virtual environment via pip in editable mode from the locally clones git repo if you want to make direct changes. Please have a look at the docs.
https://streamlink.github.io/developing.html

Also, plugin matcher changes require plugin URL test changes.

File: tests/plugins/test_tvp.py [hunk 2]
```diff
@@ -1,7 +1,6 @@
 from streamlink.plugins.tvp import TVP
 from tests.plugins import PluginCanHandleUrl
 
-
```
Comment (bastimeyer) [inline tests/plugins/test_tvp.py:4]: Don't remove this blank line. It's here deliberately.

- https://github.com/streamlink/streamlink/actions/runs/10132998278/job/28017929129?pr=6097#step:5:23

File: tests/plugins/test_tvp.py [hunk 3]
```diff
@@ -33,6 +32,11 @@ class TestPluginCanHandleUrlTVP(PluginCanHandleUrl):
             ("tvp_info", "https://www.tvp.info/78213165/euro-2024-polska-holandia-12-bartosz-salamon-czuje-sie-winny-za-te-porazke-wideo"),
             {},
         ),
+        # sport.tvp
+        (
+            ("tvp_sport", "https://sport.tvp.pl/79514191/paryz-2024-magda-linette-mirra-andriejewa-1-runda-na-zywo-transmisja-online-live-stream-igrzyska-olimpijskie-2872024"),
+            {},
+        )
```
Comment (bastimeyer) [inline tests/plugins/test_tvp.py:39]:
```suggestion
        # sport
        (
            ("tvp_sport", "https://sport.tvp.pl/79514191/paryz-2024-magda-linette-mirra-andriejewa-1-runda-na-zywo-transmisja-online-live-stream-igrzyska-olimpijskie-2872024"),
            {"stream_id": "79514191"},
        ),
```

- https://github.com/streamlink/streamlink/actions/runs/10132998279/job/28017929778?pr=6097#step:6:339
- https://github.com/streamlink/streamlink/actions/runs/10132998278/job/28017929129?pr=6097#step:5:33

**Review by bastimeyer — CHANGES_REQUESTED** [review]
Please don't forget to post a few debug logs. I don't even know if these changes cover live streams or VODs.

**Comment by czarek7711** [comment]
Thanks for the feedback and those detailed instructions.

I was in the middle of typing out this comment, when you added yours.

Please find the logs below:

> (streamlink) PS C:\Users\czys\~\venvs\streamlink> streamlink https://sport.tvp.pl/79322729/paryz-2024-kajakarstwo-gorskie-judo-tenis-na-zywo-transmisja-online-live-stream-igrzyska-olimpijskie-2872024 best --loglevel=debug
NoTagError: `git describe --long --dirty --always --tags` could not find a tag
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.12.4
[cli][debug] OpenSSL:    OpenSSL 3.0.13 30 Jan 2024
[cli][debug] Streamlink: 0.0.0+unknown
[cli][debug] Dependencies:
[cli][debug]  certifi: 2024.7.4
[cli][debug]  isodate: 0.6.1
[cli][debug]  lxml: 5.2.2
[cli][debug]  pycountry: 24.6.1
[cli][debug]  pycryptodome: 3.20.0
[cli][debug]  PySocks: 1.7.1
[cli][debug]  requests: 2.32.3
[cli][debug]  trio: 0.26.0
[cli][debug]  trio-websocket: 0.11.1
[cli][debug]  typing-extensions: 4.12.2
[cli][debug]  urllib3: 2.2.2
[cli][debug]  websocket-client: 1.8.0
[cli][debug] Arguments:
[cli][debug]  url=https://sport.tvp.pl/79322729/paryz-2024-kajakarstwo-gorskie-judo-tenis-na-zywo-transmisja-online-live-stream-igrzyska-olimpijskie-2872024
[cli][debug]  stream=['best']
[cli][debug]  --loglevel=debug
[cli][debug]  --player=D:\Programy\VLC\vlc.exe
[cli][debug]  --player-args=--file-caching=5000 --no-one-instance --play-and-exit --qt-minimal-view
[cli][debug]  --ffmpeg-ffmpeg=D:\Programy\ffmpeg\bin\ffmpeg.exe
[cli][info] Found matching plugin tvp for URL https://sport.tvp.pl/79322729/paryz-2024-kajakarstwo-gorskie-judo-tenis-na-zywo-transmisja-online-live-stream-igrzyska-olimpijskie-2872024
[utils.l10n][debug] Language code: en_US
[utils.l10n][debug] Language code: en_US
[stream.dash][debug] Available languages for DASH audio streams: pl (using: pl)
[utils.l10n][debug] Language code: en_US
[stream.dash][debug] Available languages for DASH audio streams: pl (using: pl)
[utils.l10n][debug] Language code: en_US
[utils.l10n][debug] Language code: en_US
[utils.l10n][debug] Language code: en_US
[stream.dash][debug] Available languages for DASH audio streams: pl (using: pl)
[cli][info] Available streams: 288p_dash, 396p_dash, 576p_dash, 288p_dash_alt, 396p_dash_alt, 576p_dash_alt, 720p_dash, 1080p_dash, 288p_dash_alt2, 396p_dash_alt2, 576p_dash_alt2, 720p_dash_alt, 1080p_dash_alt, 340k_alt2 (worst), 340k_alt, 340k, 288p_alt2, 288p_alt, 288p, 396p_alt2, 396p_alt, 396p, 576p_alt2, 576p_alt, 576p, 720p_alt, 720p, 1080p (best)
[cli][info] Opening stream: 1080p (hls)
[cli][info] Starting player: D:\Programy\VLC\vlc.exe
[stream.hls][debug] Reloading playlist
[cli][debug] Pre-buffering 8192 bytes
[stream.hls][debug] First Sequence: 430545851; Last Sequence: 430546075
[stream.hls][debug] Start offset: 0; Duration: None; Start Sequence: 430546073; End Sequence: None
[stream.hls][debug] Adding segment 430546073 to queue
[stream.hls][debug] Adding segment 430546074 to queue
[stream.hls][debug] Adding segment 430546075 to queue
[stream.hls][debug] Writing segment 430546073 to output
[stream.hls][debug] Segment 430546073 complete
[cli.output][debug] Opening subprocess: ['D:\\Programy\\VLC\\vlc.exe', '--input-title-format', 'https://sport.tvp.pl/79322729/paryz-2024-kajakarstwo-gorskie-judo-tenis-na-zywo-transmisja-online-live-stream-igrzyska-olimpijskie-2872024', '--file-caching=5000', '--no-one-instance', '--play-and-exit', '--qt-minimal-view', '-']
[stream.hls][debug] Writing segment 430546074 to output
[stream.hls][debug] Segment 430546074 complete
[stream.hls][debug] Writing segment 430546075 to output
[stream.hls][debug] Segment 430546075 complete
[cli][debug] Writing stream to output
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 430546076 to queue
[stream.hls][debug] Writing segment 430546076 to output
[stream.hls][debug] Segment 430546076 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 430546077 to queue
[stream.hls][debug] Writing segment 430546077 to output
[stream.hls][debug] Segment 430546077 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 430546078 to queue
[stream.hls][debug] Writing segment 430546078 to output
[stream.hls][debug] Segment 430546078 complete
[stream.segmented][debug] Closing worker thread
[stream.segmented][debug] Closing writer thread
[stream.segmented][debug] Closing writer thread
[cli][info] Stream ended
Interrupted! Exiting...
[cli][info] Closing currently open stream...
(streamlink) PS C:\Users\czys\~\venvs\streamlink>

File: tests/plugins/test_tvp.py (hunk 2 above)
Comment (czarek7711) [reply tests/plugins/test_tvp.py]: I run the ruff check.

File: tests/plugins/test_tvp.py (hunk 3 above)
Comment (czarek7711) [reply tests/plugins/test_tvp.py]: I run the pytest.

**Review by bastimeyer — APPROVED** [review]

**Comment by bastimeyer** [comment]
Thanks

**Pull request merged**